raft/testdata/campaign_learner_must_vote.txt

```diff
@@ -0,0 +1,123 @@
+# Regression test that verifies that learners can vote. This holds only in the
+# sense that if a learner is asked to vote, a candidate believes that they are a
+# voter based on its current config, which may be more recent than that of the
+# learner. If learners which are actually voters but don't know it yet don't
+# vote in that situation, the raft group may end up unavailable despite a quorum
+# of voters (as of the latest config) being available.
+#
+# See:
+# https://github.com/etcd-io/etcd/pull/10998
+
+# Turn output off during boilerplate.
+log-level none
+----
+ok
+
+add-nodes 3 voters=(1,2) learners=(3) index=2
+----
+ok
+
+campaign 1
+----
+ok
+
+stabilize
+----
+ok (quiet)
+
+propose-conf-change 1
```

Comment:
Add a comment here. I spent longer than I'd like to admit wondering where n3 was becoming a voter.

I don't know whether there's necessarily anything to do, but it is _very_ easy to get the terms `v1` and `v2` confused between their meanings as voters 1 and 2 and conf change versions 1 and 2.